Userland/Utilities/strace.cpp

```diff
@@ -36,24 +137,328 @@ static void handle_sigint(int)
+    }
+
+    StringBuilder m_builder;
+    bool m_first_arg { true };
+};
+
+static void format_open(FormattedSyscallBuilder& builder, Syscall::SC_open_params* params_p)
+{
+    auto params = copy_from_process(params_p);
+
+    if (params.dirfd == AT_FDCWD)
+        builder.add_argument("AT_FDCWD");
+    else
+        builder.add_argument(params.dirfd);
+
+    builder.add_string_argument(params.path);
+
+    Vector<StringView> active_flags;
+    if (params.options & O_RDWR)
+        active_flags.append("O_RDWR");
+    else if (params.options & O_RDONLY)
+        active_flags.append("O_RDONLY");
+    else if (params.options & O_WRONLY)
+        active_flags.append("O_WRONLY");
+
+    if (params.options & O_APPEND)
+        active_flags.append("O_APPEND");
+    if (params.options & O_CREAT)
+        active_flags.append("O_CREAT");
+    // TODO: etc...
+
+    // TODO: add to FormattedSyscallBuilder
+    StringBuilder sbuilder;
+    sbuilder.join(" | ", active_flags);
+    builder.add_argument(sbuilder.to_string());
+
+    if (params.options & O_CREAT)
+        builder.add_argument("{:04o}", params.mode);
+}
+
+namespace AK {
+template<>
+struct Formatter<struct timespec> : StandardFormatter {
+    Formatter() = default;
+    explicit Formatter(StandardFormatter formatter)
+        : StandardFormatter(formatter)
+    {
+    }
+
+    void format(FormatBuilder& format_builder, struct timespec value)
+    {
+        auto& builder = format_builder.builder();
+        builder.append('{');
+        builder.appendff("tv_sec={}, tv_nsec={}", value.tv_sec, value.tv_nsec);
+        builder.append('}');
+    }
+};
+
+template<>
+struct Formatter<struct timeval> : StandardFormatter {
+    Formatter() = default;
+    explicit Formatter(StandardFormatter formatter)
+        : StandardFormatter(formatter)
+    {
+    }
+
+    void format(FormatBuilder& format_builder, struct timeval value)
+    {
+        auto& builder = format_builder.builder();
+        builder.append('{');
+        builder.appendff("tv_sec={}, tv_usec={}", value.tv_sec, value.tv_usec);
+        builder.append('}');
+    }
+};
+
+template<>
+struct Formatter<struct stat> : StandardFormatter {
+    Formatter() = default;
+    explicit Formatter(StandardFormatter formatter)
+        : StandardFormatter(formatter)
+    {
+    }
+
+    void format(FormatBuilder& format_builder, struct stat value)
+    {
+        auto& builder = format_builder.builder();
+        builder.append('{');
+        builder.appendff("st_dev={}, st_ino={}, st_mode={}, st_nlink={}, "
+                         "st_uid={}, st_gid={}, "
+                         "st_rdev={}, st_size={}, st_blksize={}, st_blocks={}, "
+                         "st_atim={}, st_mtim={}, st_ctim={}",
+            value.st_dev,
+            value.st_ino, value.st_mode, value.st_nlink, value.st_uid, value.st_gid, value.st_rdev, value.st_size, value.st_blksize, value.st_blocks, value.st_atim, value.st_mtim, value.st_ctim);
+        builder.append('}');
```

Comment:
Same as above.
Also please either consistently split the arguments into multiple lines, or just don't.

Reply:
I changed this (and others) now. I was initially assuming that the clang-format would sort this out automatically for me, but that wasn't the case and later on I forgot to bring this back into a nicer format.

This particular call is pretty long, so I made it multiline but still a few elements per line. I tried to be more or less reasonable with the rest. Let me know if you think it still needs some improvement.

Reply:
The new version looks nice :)